Docs/source/How_To/coding-style.rst

```diff
@@ -0,0 +1,366 @@
+also be separated by snakecase and describe the function's purpose. If there are multiple words in
+the second section they should be written in CamelCase. Ex
+
+.. code-block:: c
+
+    Library_Function();
+    Contactor_GetState();
+
+Descriptions for public functions should be included above the function definition in the source file and above the declaration 
+in the header file.
+
+=================
+Private Functions
+=================
+
+Code duplication should be avoided by using private functions instead. Their format is the same as public functions but their description 
+should be included in the source file, not the header file. All private functions should be ``static`` and can be implemented in one of two ways.
+
+.. code-block:: c
+
+    /**
+     * NOTE: Include anything important someone else will need to know
+     * @brief Give Description of Function
+     * @param variable Describe input parameters
+     * @return What function returns
+     */
+    static void ADC_InitDMA(void);
+
+Implementation One: Preferred for longer function blocks
+
+.. code-block:: c 
+
+    //Start of Source File
+    static void Static_FunctionOne(inone, intwo, inthree);
+
+    //Source code with calls to function
+
+    static void Static_FunctionOne(inone, intwo, inthree){
+        //code for static function
+    }
+
+Implementation Two: Preferred for shorter function blocks
+
+.. code-block:: c 
+
+    //Start of Source File
+    static void Static_FunctionOne(inone, intwo, inthree){
+        //code for static function
+    }
+
+    //Source code with calls to function
+
+These implementations should not be mixed within the same source file (i.e you cannot have some static functions defined at the start of the file
+and some at the end).
+
+Variables
+=========
+
+================
+Local Variables
+================
+
+If variables have the same type, declare them on the same line if possible. 
+Meaningful names for variables that aren't used for iterations in a loop should be written in camelCase.
+Unnecessary variables should be avoided. i.e
+
+.. code-block:: c
+    :linenos:
+
+    //NO BUENO
+    int x = function();
+    if (x) return;
+    //YES BUENO
+    if (function()) {
+        return;
+    }
+
+If the functionality of the code is affected by this (calling the same function multiple times), then
+it is fine to have a variable for the return value.
+
+================
+Global Variables
+================
+
+Variable names should be meaningful and written in CamelCase. Efforts must be made to prevent global variables from
+being used in multiple libraries. All global variables must be static variables unless the logic requires global variables
+being shared. All global variables should have comments describing their purpose (what they hold). Ex.
+
+.. code-block:: c
+    :linenos:
+
+    static OS_MUTEX VoltageMutex;
+    static uint16_t VoltageVal[NUM_BATTERY_MODULES]; //Voltage values gathered
```

Comment:
I don't like this variable name though... I think `Voltages[NUM_BATTERY_MODULES]` would be better